libs/wingc/src/type_check.rs

```diff
@@ -836,6 +836,20 @@ impl TypeRef {
 		}
 	}
 
+	// returns true if mutable type or if immutable container type contains a mutable type
+	pub fn is_deep_mutable(&self) -> bool {
+		match &**self {
+			Type::MutArray(_) => true,
+			Type::MutMap(_) => true,
+			Type::MutSet(_) => true,
+			Type::MutJson => true,
+			Type::Array(v) => v.is_deep_mutable(),
+			Type::Map(v) => v.is_deep_mutable(),
+			Type::Set(v) => v.is_deep_mutable(),
+			_ => false,
```

Comment:
Optionals?

Maybe it's worthwhile for this to be exhaustive anyways

Reply:
Great catch!